core/src/main/java/com/netflix/conductor/core/execution/WorkflowRepairService.java

```diff
@@ -0,0 +1,112 @@
+ * task or workflow state.
+ *
+ * This service expects that the underlying Queueing layer implements QueueDAO.containsMessage method. This can be controlled
+ * with Configuration.isWorkflowRepairServiceEnabled() property.
+ */
+public class WorkflowRepairService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkflowRepairService.class);
+
+    private final ExecutionDAO executionDAO;
+    private final QueueDAO queueDAO;
+    private final Configuration configuration;
+
+    private final Predicate<Task> isSystemTask = task -> WorkflowSystemTask.is(task.getTaskType());
+
+    @Inject
+    public WorkflowRepairService(
+            ExecutionDAO executionDAO,
+            QueueDAO queueDAO,
+            Configuration configuration
+    ) {
+        this.executionDAO = executionDAO;
+        this.queueDAO = queueDAO;
+        this.configuration = configuration;
+    }
+
+    /**
+     * Verify and repair if the workflowId exists in deciderQueue, and then if each scheduled task has relevant message
+     * in the queue.
+     * @param workflowId
+     * @param includeTasks
+     * @return
+     */
+    public boolean verifyAndRepairWorkflow(String workflowId, boolean includeTasks) {
+        Workflow workflow = executionDAO.getWorkflow(workflowId, includeTasks);
+        AtomicBoolean repaired = new AtomicBoolean(false);
+        repaired.set(verifyAndRepairDeciderQueue(workflow));
+        if (includeTasks) {
+            workflow.getTasks().forEach(task -> {
+                repaired.set(verifyAndRepairTask(task));
+            });
+        }
+        return repaired.get();
+    }
+
+    /**
+     * Verify and repair tasks in a workflow
+     * @param workflowId
+     */
+    public void verifyAndRepairWorkflowTasks(String workflowId) {
+        Workflow workflow = executionDAO.getWorkflow(workflowId, true);
+        workflow.getTasks().forEach(task -> verifyAndRepairTask(task));
+    }
+
+    /**
+     * Verify and fix if Workflow decider queue contains this workflowId.
+     * @param workflow
+     * @return
+     */
+    private boolean verifyAndRepairDeciderQueue(Workflow workflow) {
+        if (!workflow.getStatus().isTerminal()) {
+            String queueName = WorkflowExecutor.DECIDER_QUEUE;
+            if (!queueDAO.containsMessage(queueName, workflow.getWorkflowId())) {
+                queueDAO.push(queueName, workflow.getWorkflowId(), configuration.getSweepFrequency());
+                Monitors.recordQueueMessageRepushFromRepairService(queueName);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Verify if ExecutionDAO and QueueDAO agree for the provided task.
+     * @param task
+     * @return
+     */
+    @VisibleForTesting
+    protected boolean verifyAndRepairTask(Task task) {
+        WorkflowSystemTask workflowSystemTask = WorkflowSystemTask.get(task.getTaskType());
+        if (task.getStatus().equals(Task.Status.SCHEDULED)) {
```

Comment:
```suggestion
        if (task.getStatus() == Task.Status.SCHEDULED)) {
```
Enums can be compared using the operator and it also helps avoiding a `NullPointerException`